app/api/userv2/event_controller.go

```diff
@@ -51,3 +52,103 @@ func createEventController(c echo.Context) error {
 
 	return c.JSON(http.StatusOK, core.ApiSuccess(map[string]interface{}{}))
 }
+
+func getEventController(c echo.Context) error {
+	_, err := core.ExtractUser(c)
+	if err != nil {
+		return core.JSONApiError(c, http.StatusUnauthorized)
+	}
+
+	var eventGetDto dto.EventGet
+	if err := c.Bind(&eventGetDto); err != nil {
+		return core.JSONApiError(c, http.StatusBadRequest)
+	}
+
+	eventID, err := uuid.Parse(eventGetDto.EventID)
+	if err != nil {
+		return core.JSONApiError(c, http.StatusBadRequest)
+	}
+
+	event, err := user.GetUserService().GetEvent(c.Request().Context(), eventID)
+
+	return c.JSON(http.StatusOK, core.ApiSuccess(map[string]interface{}{
+		"Event": event,
+	}))
+}
+
+func updateEventController(c echo.Context) error {
+	_, err := core.ExtractUser(c)
+	if err != nil {
+		return core.JSONApiError(c, http.StatusUnauthorized)
+	}
+
+	var eventUpdateDto dto.EventUpdate
+	if err := c.Bind(&eventUpdateDto); err != nil {
+		return core.JSONApiError(c, http.StatusBadRequest)
+	}
+
+	orgUser, err := core.ExtractOrguser(c, eventUpdateDto.OrganizationID)
+	if err != nil {
+		return core.JSONApiError(c, http.StatusUnauthorized)
+	}
+	eventID, err := uuid.Parse(eventUpdateDto.EventID)
+	if err != nil {
+		return core.JSONApiError(c, http.StatusBadRequest)
+	}
+
+	event, err := user.GetUserService().GetEvent(c.Request().Context(), eventID)
+	if eventUpdateDto.EventName == "" {
+		eventUpdateDto.EventName = event.EventName
+	}
+	if eventUpdateDto.EventDate.IsZero() {
+		eventUpdateDto.EventDate = event.EventDate
+	}
+	if eventUpdateDto.EventDescription == "" {
+		eventUpdateDto.EventDescription = event.EventDescription
+	}
+	if eventUpdateDto.EventLocation == "" {
+		eventUpdateDto.EventName = event.EventLocation
+	}
+	if len(eventUpdateDto.EventImage) == 0 {
+		eventUpdateDto.EventImage = []byte(event.EventImageUrl)
+	} else {
+		if err := core.FormImage(c, eventUpdateDto.EventImage, "event_image"); err != nil {
+			return core.JSONApiError(c, http.StatusBadRequest)
+		}
+	}
```

Comment:
I appreciate the attempt to validate input data, but validation in this case should be handled elsewhere, and the validation should be different. We do not need to check which fields are actually provided by the user, as the user already knows what the event they're modifying has.

I.E. if the frontend wants to modify event A
```
{
 "event_name": "abcdefg",
 "event_date": "-93213-9218312"
}
```
to change the name to `"pointless database calling"`, they could just send
```
{
 "event_name": "pointless database calling",
 "event_date": "-93213-9218312"
}
```
So that we don't have to even check what fields are being provided.